src/sugar3/graphics/progressicon.py

```diff
@@ -34,66 +41,75 @@ class ProgressIcon(Gtk.DrawingArea):
 
     """
     def __init__(self, icon_name, pixel_size, stroke_color, fill_color,
-                 direction='vertical'):
+                 direction='vertical'): '''Direction becomes vertical'''
         Gtk.DrawingArea.__init__(self)
 
-        self._icon_name = icon_name
-        self._direction = direction
-        self._progress = 0
+        self._icon_name = icon_name 
+        self._direction = direction 
+        self._progress = 0 '''Initial Value of progress is zero'''
 
         self._stroke = get_surface(
             icon_name=icon_name, width=pixel_size, height=pixel_size,
             stroke_color=stroke_color,
-            fill_color=style.COLOR_TRANSPARENT.get_svg())
-
+            fill_color=style.COLOR_TRANSPARENT.get_svg())  
+       
+        ''' icon_name=icon_name means the icon_name after '=' is from get_surface, width is parsed from pixel_size of get_surface
+            similarly, height is also parsed from pixel_size of get_surface (like XXXpx), the fill_color means fill color of progress
+            icon and its color is imported from namespace style. Here, it's color is transparent. 
+        '''
```

Comment:
Instead of the above paragraph here, explain the function parameters for the __init__ function in a generic sense.